.github/workflows/singledb-version-upgrade.yml

```diff
@@ -0,0 +1,134 @@
+name: Major Version Upgrade Tests for singledb mode
+on: [push, pull_request]
+
+jobs:
+  run-babelfish-mvu-tests-singledb:
+    env:
+      env:
+      OLD_INSTALL_DIR: postgres13
+      NEW_INSTALL_DIR: postgres14
+      ENGINE_BRANCH_FROM: BABEL_1_X_DEV__PG_13_6
+      EXTENSION_BRANCH_FROM: BABEL_1_X_DEV
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Base version ${{ env.ENGINE_BRANCH_FROM }} and run prepare tests
+        id: setup-base-version
+        if: always()
+        uses: ./.github/composite-actions/setup-base-version
+        with:
+          engine_branch: ${{ env.ENGINE_BRANCH_FROM }}
+          extension_branch: ${{ env.EXTENSION_BRANCH_FROM }}
+          install_dir: ${{ env.OLD_INSTALL_DIR }}
+          migration_mode: 'single-db'
+
+      - name: Build Modified Postgres using latest version
+        id: build-modified-postgres-new
+        if: always() && steps.setup-base-version.outcome == 'success'
+        uses: ./.github/composite-actions/build-modified-postgres
+        with:
+          install_dir: ${{env.NEW_INSTALL_DIR}}
+
+      - name: Copy ANTLR
+        run: cp "/usr/local/lib/libantlr4-runtime.so.4.9.3" ~/${{env.NEW_INSTALL_DIR}}/lib/
+
+      - name: Build Extensions using latest version
+        id: build-extensions-new
+        if: always() && steps.build-modified-postgres-new.outcome == 'success'
+        uses: ./.github/composite-actions/build-extensions
+        with:
+          install_dir: ${{env.NEW_INSTALL_DIR}}
+
+      - name: Setup new data directory
+        id: setup-new-datadir
+        if: always() && steps.build-extensions-new.outcome == 'success'
+        run: |
+          cd ~
+          ~/${{env.NEW_INSTALL_DIR}}/bin/initdb -D ~/${{env.NEW_INSTALL_DIR}}/data
```

Comment:
can we make it a separate composite action to be shared by all kind of upgrades

Reply:
Resolved in the latest commit